src/lib/storage/segment_iterables/any_segment_iterable.hpp

```diff
@@ -57,6 +57,7 @@ class BaseAnySegmentIterableWrapper {
   virtual void with_iterators(const std::shared_ptr<const PosList>& position_filter,
                               const AnySegmentIterableFunctorWrapper<ValueType>& functor_wrapper) const = 0;
   virtual size_t size() const = 0;
+  virtual const BaseSegment& segment() const = 0;
```

Comment:
Grepping this PR for `.segment()` and `->segment()` did not give me an example of this being used. What do we need it for?

Reply:
True...